spec.html

```diff
@@ -224,21 +224,36 @@ <h1>ShadowRealmImportValue ( _specifierString_, _exportNameString_, _callerRealm
 				1. If _hasOwn_ is *false*, throw a *TypeError* exception.
 				1. Let _value_ be ? Get(_exports_, _string_).
 				1. Let _realm_ be _f_.[[Realm]].
-				1. Return ? GetWrappedValue(_realm_, _value_).
+				1. Return ? GetTransferrableValue(_realm_, _value_).
 			</emu-alg>
 		</emu-clause>
 
-		<emu-clause id="sec-getwrappedvalue" aoid="GetWrappedValue">
-			<h1>GetWrappedValue ( _callerRealm_, _value_ )</h1>
+		<emu-clause id="sec-gettransferrablevalue" aoid="GetTransferrableValue">
+			<h1>GetTransferrableValue ( _receiverRealm_, _value_ )</h1>
 			<emu-alg>
-				1. Assert: _callerRealm_ is a Realm Record.
+				1. Assert: _receiverRealm_ is a Realm Record.
 				1. If Type(_value_) is Object, then
-				  1. If IsCallable(_value_) is *false*, throw a TypeError exception.
-				  1. Return ! WrappedFunctionCreate(_callerRealm_, _value_).
+					1. If IsCallable(_value_) is *true*,
+						1. Return ! WrappedFunctionCreate(_receiverRealm_, _value_).
+					1. If _value_ has an [[ArrayBufferData]] internal slot, then
+						1. If IsSharedArrayBuffer(_value_) if *true*, return ? CloneSharedArrayBuffer(_value_, _receiverRealm_.[[Intrinsics]].%SharedArrayBuffer%).
+					1. throw a TypeError exception.
 				1. Return _value_.
 			</emu-alg>
 		</emu-clause>
 
+		<emu-clause id="sec-clonesharedarraybuffer" aoid="CloneSharedArrayBuffer">
+			<h1>CloneSharedArrayBuffer ( _buffer_, _constructor_ )</h1>
+			<emu-alg>
+				1. Assert: _buffer_ is an Object with a [[SharedArrayBuffer]] internal.
+				1. Let _clone_ be ? OrdinaryCreateFromConstructor(_constructor_, "%SharedArrayBuffer.prototype%", « [[ArrayBufferData]], [[ArrayBufferByteLength]] »).
```

Comment:
the second argument is meant to be a string with the default proto. This proto will be fetched from the given constructor.

Reply:
I also mentioned this in a comment [above](https://github.com/tc39/proposal-shadowrealm/pull/336#discussion_r738916440).

> usage of `"%SharedArrayBuffer.prototype%"` that doesn't need to be prefixed by the realm. The constructor argument for OrdinaryCreateFromConstructor is the important piece there and SharedArrayPrototype.prototype is not configurable, it should be safe.

In fact, it's meant to be just an string, regardless realms. This is string is gonna be used to access the matching intrinsic property of the given constructor.

Also, this name is just symbolic, as the `%SharedArrayBuffer%.prototype` will always be available:

```
OrdinaryCreateFromConstructor ( constructor, intrinsicDefaultProto [ , internalSlotsList ] )

1. Assert: intrinsicDefaultProto is this specification's name of an intrinsic object. The corresponding object must be an intrinsic that is intended to be used as the [[Prototype]] value of an object.
2. Let proto be ? GetPrototypeFromConstructor(constructor, intrinsicDefaultProto).
3. Return ! OrdinaryObjectCreate(proto, internalSlotsList).

GetPrototypeFromConstructor ( constructor, intrinsicDefaultProto )

1. Assert: intrinsicDefaultProto is this specification's name of an intrinsic object. The corresponding object must be an intrinsic that is intended to be used as the [[Prototype]] value of an object.
2. Let proto be ? Get(constructor, "prototype").
3. If Type(proto) is not Object, then
  a. Let realm be ? GetFunctionRealm(constructor).
  b. Set proto to realm's intrinsic object named intrinsicDefaultProto.
4. Return proto.
```

The name would only be used in step 3.b for the abstraction above, but `%SharedArrayBuffer%.prototype` is non configurable, non writable and the constructor is always the intrinsic one.

That means `OrdinaryCreateFromConstructor` results in `OrdinaryObjectCreate(_receiverRealm_.[[Intrinsics]].%SharedArrayBuffer.prototype%, « [[ArrayBufferData]], [[ArrayBufferByteLength]] »)`.

I can actually shift usage from OrdinaryCreateFromConstructor to OrdinaryObjectCreate. The current draft uses it to match the slice method in SharedArrayBuffer.

Reply:
The other reason for using `OrdinaryCreateFromConstructor` is only if the new abstraction `CloneArrayBuffer` becomes reusable within ECMAScript, but apparently I need to avoid this new abstraction as it became noise here.